tests/cxx/placement_new-d1.h

```diff
@@ -12,4 +12,7 @@
 
 #include "tests/cxx/placement_new-i1.h"
 
+// for std::addressof use in .cc
+#include <utility>
```

Comment:
It is stated in the standard that `std::addressof` is in `<memory>`:
https://eel.is/c++draft/memory.syn
Hard-coded symbol mapping should be probably updated.

Reply:
Yeah, I saw that too. There's a header mapping from `bits/stl_move.h`, where `std::addressof` is defined, to `<utility>`, presumably for `std::move`. I'd rather not mess with this right now.

But maybe it would be nicer to get rid of the standard library dependency in the test; this function is easy enough to simulate.

Reply:
Added `placement_new-i2.h` with a custom `AddressOf` function.